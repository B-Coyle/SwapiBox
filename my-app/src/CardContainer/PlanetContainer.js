import React, { Component } from "react";
import Card from "../Card/Card";
import {
  fetchMaster,
  fetchResidents
} from "../ApiCall/apiCalls.js";
const uuidv4 = require("uuid/v4");

class PlanetsContainer extends Component {
  constructor() {
    super();
    this.state = {
      planets: [],
      error: "",
      isLoading: false
    };
  }

  componentDidMount() {
    this.setState({ 
      isLoading: true 
    });

    this.setPlanets();
  }

  setPlanets = () => {
    fetchMaster('planets')
      .then(data => fetchResidents(data.results))
      .then(planets => this.formatPlanet(planets));
  };

  formatPlanet = planets => {
    let trimmedPlanets = planets.map(planet => {
       return planet.length
      ? planet.pop() 
      : planet
    });
    let updatedPlanets = trimmedPlanets.map(planet => {
      let { name, terrain, population, climate, residentNames } = planet;
      return { name, terrain, population, climate, residents: residentNames, key: uuidv4() };
    });
    this.setState({ 
      planets: updatedPlanets, 
      isLoading: false 
    });
  };

  render() {
    const planetCards = this.state.planets.map(planet => (
      <Card key={planet.key} data={planet} category="planet" />
    ));

    const loadingMessage = (
      <div className="loadingMessage">
        <p>Information loading. Please Hold...</p>
      </div>
    );
    return (
      <section className="cardContainer planetContainer">
        {this.state.isLoading ? loadingMessage : planetCards}
      </section>
    );
  }
}

export default PlanetsContainer;
