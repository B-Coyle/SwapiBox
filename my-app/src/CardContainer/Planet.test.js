import React from 'react';
import PlanetContainer from '../CardContainer/PlanetContainer.js';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';


describe('PlanetContainer', () => {
    let wrapper; 
    const defaultState ={
            planets: [],
            error: "",
            loading: false
          }
    beforeEach(()=> {
    wrapper = shallow(<PlanetContainer/>);

    });
    it('should match the snapshot with all data passed in', () => {
        expect(wrapper).toMatchSnapshot();
        });
    
    it("should have default state", () => {
        expect(wrapper.state()).toEqual(defaultState);
    });

})