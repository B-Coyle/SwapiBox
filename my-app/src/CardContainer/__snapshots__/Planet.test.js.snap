// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetContainer should match the snapshot with all data passed in 1`] = `
<section
  className="cardContainer planetContainer"
>
  <div
    className="loadingMessage"
  >
    <p>
      Information loading. Please Hold...
    </p>
  </div>
</section>
`;
